lib/stream.js

```diff
@@ -1,11 +1,14 @@
 'use strict';
 
-module.exports = Stream;
+var Stream = require('internal/streams/stream').Stream;
+
+// wrap the old-style stream
+require('internal/streams/legacy')(Stream);
```

Comment:
This does not make a lot of sense to me. Basically the definition of `Stream` is divided into two files.
Not good in my opinion. If we also move the definition of `Stream`, I think `internal/streams/stream` should also include the 'legacy' bits.

How about we move all `_stream` prefixed files into `internal/streams`? That might be a better solution overall.
cc @nodejs/streams.